File: Week 2/C Programming/prog3.c
#include <stdio.h>

int main() {
    int n;
    printf("Enter size of square matrix: ");
    scanf("%d", &n);

    int mat[n][n];
    printf("Enter matrix elements:\n");
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            scanf("%d", &mat[i][j]);
        }
    }

    int primarySum = 0, secondarySum = 0;
    for (int i = 0; i < n; i++) {
        primarySum += mat[i][i];               // Primary diagonal
        secondarySum += mat[i][n - i - 1];     // Secondary diagonal
    }

    printf("Primary diagonal sum = %d\n", primarySum);
    printf("Secondary diagonal sum = %d\n", secondarySum);

    // If you want total diagonal sum (without double counting middle element in odd n)
    int totalSum = primarySum + secondarySum;
    if (n % 2 != 0) {
        totalSum -= mat[n / 2][n / 2]; // Remove middle element counted twice
    }
    printf("Total diagonal sum = %d\n", totalSum);

    return 0;
}

File: Week 2/python Programming/prog3.py
def sum_of_diagonals(matrix):
    n = len(matrix)
    primary_sum = 0
    secondary_sum = 0

    for i in range(n):
        primary_sum += matrix[i][i]             # Primary diagonal
        secondary_sum += matrix[i][n - i - 1]   # Secondary diagonal

    # If matrix size is odd, subtract the middle element once (counted twice)
    total_sum = primary_sum + secondary_sum
    if n % 2 != 0:
        total_sum -= matrix[n // 2][n // 2]

    return primary_sum, secondary_sum, total_sum

# Example:
matrix = [
    [1, 2, 3],
    [4, 5, 6],
    [7, 8, 9]
]

p_sum, s_sum, t_sum = sum_of_diagonals(matrix)
print(f"Primary diagonal sum = {p_sum}")
print(f"Secondary diagonal sum = {s_sum}")
print(f"Total diagonal sum = {t_sum}")
